### PR TITLE
Implement comparison methods

### PR DESCRIPTION
**Pull request by SivanMehta**
This PR implements the `compareHeight` and the `compareWidth` methods. Here is the documentation added to the README that describes the usage of these methods:

### `compareWidth(breakpoint)`

Returns the difference between the current width and the given breakpoint. This can be used to check if the window is "greater" than a breakpoint. If the given breakpoint does not have a width, this will always return the current width. If the given breakpoint does not exist, than we return an error.

```js
var breakpoints = new Breakdancer([
  {
    name: 'wrist',
    width: 320
  },
  {
    name: 'palm',
    width: 800,
    height: 600
  }
]);

// let's assume the window 500 is px wide.
console.log(breakpoints.compareWidth('wrist')) // 180
console.log(breakpoints.compareWidth('palm')) // -300
```

### `compareHeight(breakpoint)`

Same usage as `compareWidth`

# TL;DR

Now you can check if you're "past" a certain breakpoint with `compareWidth('mobile')`

### CODE REVIEW

File: test.js [hunk 1]
```diff
@@ -168,4 +169,41 @@ describe('breakdancer', function () {
...
+  });
+
+  describe('#compare{Height,Width}', function () {
+    var bd = new Breakdancer(specification);
+    var spy = sinon.spy(bd, 'compare');
+
+    it('should call compare', function () {
+      bd.compareHeight('mobile');
+      bd.compareWidth('mobile');
+      assume(spy.callCount).equals(2);
```
Comment (Swaagie) [inline test.js]: might want to assert the functions are called with the right property e.g. `width` and `height`

File: breakdancer.js [hunk 2]
```diff
@@ -94,4 +94,54 @@ export default class Breakdancer {
 
     return 'unknown';
   }
+
+  /**
+   * Compare current and specified properties of given breakpoint
+   * Requires that this.height() and this.width() are implemented
```
Comment (Swaagie) [inline breakdancer.js]: wonder if we should safeguard against this, although both native and regular have default implementations.

Comment (SivanMehta) [reply breakdancer.js]: I was thinking that because this is already a `@private` method, we can safeguard this by relying on the fact `this.height()` and `this.width()` methods **already exist** in `index.js` and `index.native.js`.

However, it would be worthwhile to throw a `ReferenceError` if the associated `this[property]` function did not exist.

File: breakdancer.js [hunk 3]
```diff
@@ -94,4 +94,54 @@ export default class Breakdancer {
...
+   *
+   * @param {String} breakpoint to be compared
+   * @param {String} property height or width
+   * @throws {TypeError} If not given breakpoint and property do not exist within the given spec
+   * @returns {Number} different between current and specified properties
+   * @private
```
Comment (3rd-Eden) [inline breakdancer.js]: In all honestly, I would say you might as well just ditch `compareHeight` and `compareWidth` and make the `compare` method public and keep that as only API.
